yewprint/src/spinner.rs

```diff
@@ -0,0 +1,111 @@
+use crate::Intent;
+use std::cmp::min;
+use yew::prelude::*;
+
+pub const R: i32 = 45;
+pub const PATH_LENGTH: i32 = 280;
+pub const MIN_SIZE: i32 = 10;
+pub const STROKE_WIDTH: i32 = 4;
+pub const MIN_STROKE_WIDTH: i32 = 16;
+pub const SIZE_SMALL: i32 = 20;
+pub const SIZE_STANDARD: i32 = 50;
+pub const SIZE_LARGE: i32 = 100;
+
+pub struct Spinner {
+    props: SpinnerProps,
+}
+
+#[derive(Clone, PartialEq, Properties)]
+pub struct SpinnerProps {
+    #[prop_or_default]
+    pub intent: Option<Intent>,
+    #[prop_or_default]
+    pub class: Classes,
+    #[prop_or_default]
+    pub size: i32,
+    #[prop_or_default]
+    pub value: Option<f32>,
+}
+
+impl Component for Spinner {
+    type Message = ();
+    type Properties = SpinnerProps;
+
+    fn create(props: Self::Properties, _link: ComponentLink<Self>) -> Self {
+        Spinner { props }
+    }
+
+    fn update(&mut self, _msg: Self::Message) -> ShouldRender {
+        true
+    }
+
+    fn change(&mut self, props: Self::Properties) -> ShouldRender {
+        if self.props != props {
+            self.props = props;
+            true
+        } else {
+            false
+        }
+    }
+
+    fn view(&self) -> Html {
+        let default_value = 0.25;
+        let path_length_float = PATH_LENGTH as f32;
+        let size = self.props.size;
+        let stroke_width = min(MIN_STROKE_WIDTH, (STROKE_WIDTH * SIZE_LARGE) / size);
+
+        let view_box = {
+            let radius = R as f32 + stroke_width as f32 / 2.00;
+            let view_box_x = 50.00 - radius;
+            let view_box_width = radius * 2.00;
+            format!(
+                "{:.2} {:.2} {:.2} {:.2}",
+                view_box_x, view_box_x, view_box_width, view_box_width,
+            )
+        };
+        let spinner_track: String = format!(
+            "M 50,50 m 0,-{R} a {R},{R} 0 1 1 0,{R2} a {R},{R} 0 1 1 0,-{R2}",
+            R = R,
+            R2 = R * 2,
+        );
+        let stroke_offset = path_length_float
+            - path_length_float
+                * (self
+                    .props
+                    .value
+                    .map(|x| x.clamp(0.0, 1.0))
+                    .unwrap_or(default_value));
+        html! {
```

Comment:
```suggestion
                    .unwrap_or(default_value));

        html! {
```

Space before the ending statement